src/vsr/journal.zig

```diff
@@ -1643,6 +1643,8 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
             assert(journal.has_dirty(message.header));
 
             const write = journal.writes.acquire() orelse {
+                assert(replica.replica_count > 1);
```

Comment:
This was already true, just adding an assert because I got tripped up on this at one point.

`replica_count=1` serializes its WAL writes writes (unlike all other cluster sizes which allow concurrent writes).
The reason is that otherwise `replica_count=1` risk forking the log if they ever lose a write (e.g. in the middle of a pipeline) followed by a crash/restart. Of course cluster-of-1 can't recover from a lost write, but this downgrades it from a correctness issue to an availability issue.

An alternate approach would be: Don't have special cases to serialize WAL writes in the replica, and instead hard-code `replica_count=1`'s pipeline to be 1 entry. But that would make a comptime parameter (pipeline size) depend on a runtime parameter (cluster size), which is annoying, and also I hadn't thought of that approach until just now...